newIDE/app/src/ProjectsStorage/ResourceFetcher/index.js

```diff
@@ -39,14 +57,47 @@ type UseResourceFetcherOutput = {|
    * and must optionally be fetched by the storage provider (e.g: a URL to be downloaded).
    */
   ensureResourcesAreFetched: (
-    options: FetchAllProjectResourcesOptionsWithoutProgress
+    options: CleanAndFetchAllProjectResourcesOptionsWithoutProgress
   ) => Promise<void>,
   /**
    * Render, if needed, the dialog that will show the progress of resources fetching.
    */
   renderResourceFetcherDialog: () => React.Node,
 |};
 
+export const cleanUpResourceNames = ({
+  project,
+}: CleanAndFetchAllProjectResourcesOptionsWithoutProgress) => {
+  const resourcesManager = project.getResourcesManager();
+  const allResourceNames = resourcesManager.getAllResourceNames().toJSArray();
+  const resourceToFetchNames = allResourceNames.filter(resourceName => {
+    const resource = resourcesManager.getResource(resourceName);
+
+    return isFetchableUrl(resource.getFile());
+  });
+
+  resourceToFetchNames.forEach(resourceName => {
+    const resource = resourcesManager.getResource(resourceName);
+    const resourceFile = resource.getFile();
+    let cleanedUpFilename;
+    if (isProductAuthorizedResourceUrl(resourceFile)) {
+      // Resource is a private asset.
+      cleanedUpFilename = extractFilenameWithExtensionFromProductAuthorizedUrl(
+        resourceFile
+      );
+    } else if (isPublicAssetResourceUrl(resourceFile)) {
+      // Resource is a public asset.
+      cleanedUpFilename = extractFilenameWithExtensionFromPublicAssetResourceUrl(
+        resourceFile
+      );
+    } else {
+      // Resource is a generic url.
+      cleanedUpFilename = path.basename(resourceFile);
```

Comment:
Could we just have this being factored into a function, and reused whenever need in the various resource fetchers?